mypy/version.py

```diff
@@ -9,22 +10,21 @@
 __version__ = '0.940+dev'
 base_version = __version__
 
-# tuple[major, minor, patch, releaselevel, mypy version, mypy releaselevel, hash]
-__based_version_info__ = (
-    2,
-    0,
+based_version_info = VersionInfo(
+    1,
+    4,
     0,
     "dev",
+    0,
     __version__.split("+dev")[0],
-    "dev" if "+dev" in __version__ else "release",
-    None,
+    "dev" if "+dev" in __version__ else "final",
 )
-
-__based_version__ = ".".join(str(e) for e in __based_version_info__[:3])
+__based_version__ = based_version_info.simple_str()
+base_based_version = __based_version__
```

Comment:
can u add documentation or better names for these version variables? currently looks pretty confusing